google/auth/credentials.py

```diff
@@ -236,6 +223,40 @@ def with_non_blocking_refresh(self):
         self._use_non_blocking_refresh = True
 
 
+class CredentialsWithTrustBoundary(Credentials):
+    """Abstract base for credentials supporting trust boundary factory"""
```

Comment:
nit: supporting trust boundary factory? Or trust boundaries in general? Also can you add a short explanation of what trust boundaries are?